### PR TITLE
UI Fixes for Tool Interfaces

### PR DESCRIPTION
**Pull request by vlagrassa**

### CODE REVIEW

File: src/modules/site-v2/templates/_scripts/gene-search.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  gene = gene.trim();
+
+  // Get the body of the table to fill out
+  const tbody = $(`${table_id} > tbody`);
+  console.log(tbody);
+  
+  // If no search term provided, hide both dropdown elements
+  if (gene.length == 0) {
+    $(table_id).fadeOut();
+    $(loading_id).fadeOut();
```
Comment (r-vieira) [inline src/modules/site-v2/templates/_scripts/gene-search.js:47]: if this code branch is done, consider returning.

File: src/modules/site-v2/templates/_scripts/gene-search.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  
+  // If search provided, query the database
+  else {
+    tbody.html("");
+    $.ajax({
+      url:         "{{ url_for('api_gene.api_search_genes', query='') }}" + gene + '?species=' + species.name,
+      method:      "GET",
+      contentType: 'application/xml',
+    }).done(function(results) {
```
Comment (r-vieira) [inline src/modules/site-v2/templates/_scripts/gene-search.js:57]: handle the promise exception/error (API issues, etc).

File: src/modules/site-v2/templates/tools/variant_annotation/vbrowser.html [hunk 3]
```diff
@@ -261,6 +262,17 @@
...
 
+  // Enable or disable form inputs that depend on species being defined
+  const form_ids = [
+    'gene-search',
+    'interval',
+    'search-interval-btn',
+    'strain-filter',
+  ]
+  form_ids.forEach((form_id) => {
+    document.getElementById(form_id).disabled = !species_id;
```
Comment (r-vieira) [inline src/modules/site-v2/templates/tools/variant_annotation/vbrowser.html:273]: this is legacy code, but can we clarify the semantics of `!species_id`?

**Review by r-vieira — CHANGES_REQUESTED** [review]
Looks Great! See the promise error/exception handling and minor notes.

File: src/modules/site-v2/templates/_scripts/gene-search.js (hunk 1 above)
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/gene-search.js:46]: Fixed in 933a68e8db7389c88011071a364ff72777a3c451

File: src/modules/site-v2/templates/tools/variant_annotation/vbrowser.html (hunk 3 above)
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/variant_annotation/vbrowser.html]: Fixed in 0ef91f28589e0a20241e7923fa789b5da1f35672

File: src/modules/site-v2/templates/_scripts/gene-search.js (hunk 2 above)
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/gene-search.js]: Added a basic `console.error` handler in e0c14739b15ba3dea8061a3d82005aa5081cfbfb.  More rigorous error handling will be a bit tricky:

- Currently, we display errors to the user by flashing them in Flask and reloading the page -- this is a bit hard to replicate in JS.
- If we want to display to the user, we'll have to work out what an appropriate error message would be.

What would you recommend for handling errors from this call?

**Review by r-vieira — APPROVED** [review]
Looks great!

**Pull request merged**